Code/GraphMol/Wrap/SubstanceGroup.cpp

```diff
@@ -70,6 +100,48 @@ struct sgroup_wrap {
             "GetBonds", &SubstanceGroup::getBonds,
             "returns a list of the indices of the bonds in this SubstanceGroup",
             python::return_value_policy<python::copy_const_reference>())
+        .def("AddAtomWithIdx", &SubstanceGroup::addAtomWithIdx)
+        .def("AddBondWithIdx", &SubstanceGroup::addBondWithIdx)
+        .def("AddParentAtomWithIdx", &SubstanceGroup::addParentAtomWithIdx)
+        .def("AddAtomWithBookmark", &SubstanceGroup::addAtomWithBookmark)
+        .def("AddParentAtomWithBookmark",
+             &SubstanceGroup::addParentAtomWithBookmark)
+        .def("AddCState", &SubstanceGroup::addCState)
+        .def("AddBondWithBookmark", &SubstanceGroup::addBondWithBookmark)
+        .def("AddAttachPoint", &SubstanceGroup::addAttachPoint)
+        .def("AddBracket", addBracketHelper)
+
+        .def("SetProp",
+             (void (RDProps::*)(const std::string &, std::string, bool) const) &
+                 SubstanceGroup::setProp<std::string>,
+             (python::arg("self"), python::arg("key"), python::arg("val"),
+              python::arg("computed") = false),
+             "sets the value of a particular property")
+        .def("SetDoubleProp",
+             (void (RDProps::*)(const std::string &, double, bool) const) &
+                 SubstanceGroup::setProp<double>,
+             (python::arg("self"), python::arg("key"), python::arg("val"),
+              python::arg("computed") = false),
+             "sets the value of a particular property")
+        .def("SetIntProp",
+             (void (RDProps::*)(const std::string &, int, bool) const) &
+                 SubstanceGroup::setProp<int>,
+             (python::arg("self"), python::arg("key"), python::arg("val"),
+              python::arg("computed") = false),
+             "sets the value of a particular property")
+        .def(
+            "SetUnsignedProp",
+            (void (RDProps::*)(const std::string &, unsigned int, bool) const) &
+                SubstanceGroup::setProp<unsigned int>,
+            (python::arg("self"), python::arg("key"), python::arg("val"),
+             python::arg("computed") = false),
+            "sets the value of a particular property")
+        .def("SetBoolProp",
+             (void (RDProps::*)(const std::string &, bool, bool) const) &
+                 SubstanceGroup::setProp<bool>,
+             (python::arg("self"), python::arg("key"), python::arg("val"),
+              python::arg("computed") = false),
+             "sets the value of a particular property")

```

Comment:
Missing GetPropNames, GetPropsAsDict
```
        .def("GetPropNames", &SubstanceGroup::getPropList,
             (python::arg("self"), python::arg("includePrivate") = false,
              python::arg("includeComputed") = false),
             "Returns a tuple with all property names for this molecule.\n\n"
             "  ARGUMENTS:\n"
             "    - includePrivate: (optional) toggles inclusion of private "
             "properties in the result set.\n"
             "                      Defaults to 0.\n"
             "    - includeComputed: (optional) toggles inclusion of computed "
             "properties in the result set.\n"
             "                      Defaults to 0.\n\n"
             "  RETURNS: a tuple of strings\n")

        .def("GetPropsAsDict", GetPropsAsDict<SubstanceGroup>,
             (python::arg("self"), python::arg("includePrivate") = false,
              python::arg("includeComputed") = false),
             "Returns a dictionary populated with the molecules properties.\n"
             " n.b. Some properties are not able to be converted to python "
             "types.\n\n"
             "  ARGUMENTS:\n"
             "    - includePrivate: (optional) toggles inclusion of private "
             "properties in the result set.\n"
             "                      Defaults to False.\n"
             "    - includeComputed: (optional) toggles inclusion of computed "
             "properties in the result set.\n"
             "                      Defaults to False.\n\n"
             "  RETURNS: a dictionary\n")
```

Reply:
We should just make an RDProp base class and get rid of this stuff :)

Reply:
`GetPropNames()` and `GetPropsAsDict()` are already there: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/Wrap/SubstanceGroup.cpp#L97

Agreed, making an RDProp base class would simplify the wrappers a lot. Definitely something we should do for the next release.